pep-0394.txt

```diff
@@ -5,104 +5,181 @@ Last-Modified: $Date$
-  command; see the `Rationale`_ and `Migration Notes`_ below).
-* The Python 2.x ``idle``, ``pydoc``, and ``python-config`` commands should
-  likewise be available as ``idle2``, ``pydoc2``, and ``python2-config``,
-  with the original commands invoking these versions by default, but possibly
-  invoking the Python 3.x versions instead if configured to do so by the
+* If the ``python`` command is installed, it should either invoke the same
+  version of Python as the ``python3`` command or as the ``python2``
+  command.
+* Distributors may choose to set the behavior of the ``python`` command,
+  not provide it at all, or make it configurable by the user or
   system administrator.
-* In order to tolerate differences across platforms, all new code that needs
-  to invoke the Python interpreter should not specify ``python``, but rather
-  should specify either ``python2`` or ``python3`` (or the more specific
-  ``python2.x`` and ``python3.x`` versions; see the `Migration Notes`_).
-  This distinction should be made in shebangs, when invoking from a shell
-  script, when invoking via the system() call, or when invoking in any other
-  context.
-* One exception to this is scripts that are deliberately written to be source
-  compatible with both Python 2.x and 3.x. Such scripts may continue to use
-  ``python`` on their shebang line.
-* When packaging software that is source compatible with both versions,
-  distributions may change such ``python`` shebangs to ``python3``.
-  This ensures software is used with the latest version of
-  Python available, and it can remove a dependency on Python 2.
+* The Python 3.x ``idle``, ``pydoc``, and ``python-config`` commands should
+  likewise be available as ``idle3``, ``pydoc3``, and ``python3-config``;
+  Python 2.x versions as ``idle2``, ``pydoc2``, and ``python2-config``.
+  The original commands should either invoke the same version of Python
+  as the ``python`` command, or not be available at all.
+* When packaging software, distributors are encouraged to change less specific
+  shebangs to more specific ones. This ensures software is used with the latest
+  version of Python available, and it can remove a dependency on Python 2.
+  The specifics are left on the distributors.
+  Examples of this include changing ``python`` shebangs to ``python3``
+  when Python 3.x is supported, changing ``python`` shebangs to ``python2``
+  when Python 3.x is not yet supported, or changing ``python3`` shebangs to
+  ``python3.8`` if the software is built with Python 3.8.
+* When a virtual environment (created by the PEP 405 ``venv`` package or a
+  similar tool such as ``virtualenv`` or ``conda``) is active, the ``python``
+  command should refer to the virtual environment's interpreter and should
+  always be available.
+  The ``python3`` or ``python2`` command (according to the environment's
+  interpreter) should also be available.
+
+For developers
```

Comment:
Do we want to split this into script developers/maintainers and script users (who may not be the script's author)?

Reply:
There aren't any recommendations for script users (unless you count "You may need to edit shebang lines to get scripts to work on your system" as a recommendation)

Reply:
I think that it's worth, even though the recommendation is short, to split out script end users from script developers.